### PR TITLE
(MODULES-1886) Add ability to specify server_config location

### PR DESCRIPTION
**Pull request by cyberious**

### CODE REVIEW

**Comment by mhaskel** [comment]
@cyberious should have README update.

File: manifests/config/server.pp [hunk 1]
```diff
@@ -61,10 +62,15 @@
 
   $changes = delete_undef_values([$_class_name, $_address, $_port, $_shutdown])
 
+  $_server_config_location = $server_config ?  {
+    undef => "${catalina_base}/conf/server.xml",
```
Comment (bmjen) [inline manifests/config/server.pp]: space rocket

File: spec/defines/config/server_spec.rb [hunk 2]
```diff
@@ -35,6 +35,27 @@
...
     }
   end
+  context 'custom server_config location' do
+    let(:params){ {
+      :catalina_base => '/opt/apache-tomcat/test',
+      :class_name => 'foo',
+      :address => 'localhost',
+      :port => '8005',
+      :shutdown => 'SHUTDOWN',
+      :server_config => '/opt/apache-tomcat/server.xml'
```
Comment (mhaskel) [inline spec/defines/config/server_spec.rb]: align these too please.

File: spec/defines/config/server_spec.rb [hunk 3]
```diff
@@ -35,6 +35,27 @@
...
+    }}
+    it { is_expected.to contain_augeas('server-/opt/apache-tomcat/test').with(
+                          'lens' => 'Xml.lns',
+                          'incl' => '/opt/apache-tomcat/server.xml',
+                          'changes' => [
```
Comment (mhaskel) [inline spec/defines/config/server_spec.rb]: and these

**Comment by mhaskel** [comment]
This change needs to be bigger to actually solve the problem. The server.xml file is hardcoded in all the classes under tomcat::config::server::*

**Comment by mhaskel** [comment]
Closing as this was pulled in to #82

**Pull request closed without merging**